crates/cxx-qt/src/lib.rs

```diff
@@ -31,3 +31,11 @@ pub trait CxxQtType {
     /// The property changed signal must be emitted manually.
     unsafe fn rust_mut(self: core::pin::Pin<&mut Self>) -> core::pin::Pin<&mut Self::Rust>;
 }
+
+/// Types which have Locking implemented in the C++ generation
+//
+// This could be implemented using an auto trait in the future once stable
+// https://doc.rust-lang.org/beta/unstable-book/language-features/auto-traits.html
+pub unsafe trait Locking {
```

Comment:
Should the trait itself be `unsafe` ?

Reply:
Maybe yes as we want the developer to write `unsafe impl cxx_qt::Locking`.

Then this matches `Send` and `Sync` in Rust that are unsafe traits and then have an unsafe impl internally.

Reply:
Changed to a normal trait that you need to unsafe impl via our macros.